src/interface/meshblock_data.cpp

```diff
@@ -424,149 +423,6 @@ void MeshBlockData<T>::Remove(const std::string &label) {
   throw std::runtime_error("MeshBlockData<T>::Remove not yet implemented");
 }
 
-template <typename T>
-void MeshBlockData<T>::SetLocalNeighborAllocated() {
```

Comment:
Its fine to delete all this stuff, but it's worth noting that this makes it harder to write meshblockdata-meshdata agnostic code. I think the cleanup is worth it though.

Reply:
We should think if the philosophy going forward is `MeshBlockData`-`MeshData` agnostic code or all `MeshData` code and just have single block `MeshData`.

Reply:
yes---agreed. I think `MeshBlockData`/`MeshData` agnostic code was always a transition plan as we move towards all `MeshData` all the time.